jvm/ml4ir-inference/src/main/java/ml4ir/inference/tensorflow/utils/ModelFeatures.java

```diff
@@ -0,0 +1,158 @@
+package ml4ir.inference.tensorflow.utils;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModelFeatures {
+    private QueryKey query_key;
+    private Rank rank;
+    private Label label;
+    private List<InputFeatures> features;
+
+    public ModelFeatures() {}
+
+    public QueryKey getQuery_key() {
+        return query_key;
+    }
+
+    public Rank getRank() {
+        return rank;
+    }
+
+    public Label getLabel() {
+        return label;
+    }
+
+    public List<InputFeatures> getFeatures() {
+        return features;
+    }
+
+    public Map<String, Map<String, String>> getDataTypesForFeatures() {
+        Map<String, Map<String, String>> data = new HashMap<>();
+        data.put(this.getLabel().getTfrecord_type(),
+                ImmutableMap.of(this.getLabel().getName(), this.getLabel().getDtype()));
+        data.put(this.getRank().getTfrecord_type(),
+                ImmutableMap.of(this.getRank().getName(), this.getRank().getDtype()));
+        data.put(this.getQuery_key().getTfrecord_type(),
+                ImmutableMap.of(this.getQuery_key().getName(), this.getLabel().getDtype()));
+        features.forEach(f -> data.put(f.getTfrecord_type(),
+                ImmutableMap.of(f.getName(), f.getDtype())));
+        return data;
+    }
+
+    public  Map<String, Map<String, String>> getDefaultValuesForFeatures() {
+        Map<String, Map<String, String>> data = new HashMap<>();
+        data.put(this.getLabel().getTfrecord_type(),
+                ImmutableMap.of(this.getLabel().getName(), this.getLabel().getDefault_value()));
+        data.put(this.getRank().getTfrecord_type(),
+                ImmutableMap.of(this.getRank().getName(), this.getRank().getDefault_value()));
+        data.put(this.getQuery_key().getTfrecord_type(),
+                ImmutableMap.of(this.getQuery_key().getName(), this.getQuery_key().getDefault_value()));
+        features.forEach(f -> data.put(f.getTfrecord_type(),
+                ImmutableMap.of(f.getName(), f.getDefault_value())));
+        return data;
+    }
+
+    public Map<String, Map<String, String>> getServingNameMappingForFeatures() {
+        Map<String, Map<String, String>> data = new HashMap<>();
```

Comment:
same here: just make one big `ImmutableMap.of(<key>, ImmutableMap.of(<inner key>, ...`